src/lib.rs

```diff
@@ -1,4 +1,4 @@
-#![deny(clippy::clone_on_ref_ptr)]
+#![deny(warnings, clippy::pedantic)]
```

Comment:
do you want to disable `clone_on_ref_ptr` warning? I believe it is not actually included in pedantic https://rust-lang.github.io/rust-clippy/master/#clone_on_ref_ptr